cli/src/main/java/org/jboss/as/cli/handlers/ReadAttributeHandler.java

```diff
@@ -116,8 +116,12 @@ public boolean canAppearNext(CommandContext ctx) throws CommandFormatException {
                     op.get(Util.ADDRESS).setEmptyList();
                 } else {
                     final ModelNode addrNode = op.get(Util.ADDRESS);
-                    for(OperationRequestAddress.Node node : address) {
-                        addrNode.add(node.getType(), node.getName());
+                    for (OperationRequestAddress.Node node : address) {
+                        if (!(node.getName() == null)) {
+                            addrNode.add(node.getType(), node.getName());
```

Comment:
@parsharma , in this case you are replacing the current address (that is being built) by the empty address, so you loose the context (the various cd commands the user did). In order to keep the context, you could detect that the node.getName() returns null and in this case use the "*" value for the name to express the fact that the node name has not been set.
As a side note, it would be nice if you could add a test. It should be added in this file:
https://github.com/wildfly/wildfly-core/blob/master/testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java